lib/IOExt/IOExt.cpp

```diff
@@ -3,253 +3,219 @@
 //
 
 #include "../../include/definitions.h"
-#include <I2CBus.h>
-#include <Wire.h>    // I2C
 #include "PCF8574.h" // PCF8574
+#include <I2CBus.h>
+#include <Wire.h> // I2C
 
 #include "DriverDisplay.h"
 #include "IOExt.h"
 #include "Indicator.h"
 
 #define PCF8574_NUM_PORTS 8
 
-PCF8574 pcf8574(I2C_ADDRESS_PCF8574, I2C_SDA, I2C_SCL, PCF8574_INTERRUPT_PIN, key_pressed_interrupt_handler);
+PCF8574 pcf8574(I2C_ADDRESS_PCF8574, I2C_SDA, I2C_SCL, PCF8574_INTERRUPT_PIN,
+                key_pressed_interrupt_handler);
 
 int speed = 0;
 int taskSleep = 10;
 
-void scan()
-{
-    /* I2C slave Address Scanner
-    for 5V bus
-        * Connect a 4.7k resistor between SDA and Vcc
-        * Connect a 4.7k resistor between SCL and Vcc
-    for 3.3V bus
-        * Connect a 2.4k resistor between SDA and Vcc
-        * Connect a 2.4k resistor between SCL and Vcc
-    */
-    Serial.println("Scanning I2C Addresses");
-    uint8_t cnt = 0;
-    for (uint8_t i = 0; i < 128; i++)
-    {
-        Wire.beginTransmission(i);
-        uint8_t ec = Wire.endTransmission(true);
-        if (ec == 0)
-        {
-            if (i < 16)
-                Serial.print('0');
-            Serial.print(i, HEX);
-            cnt++;
-        }
-        else
-        {
-            Serial.print("--");
-        }
-        Serial.print(' ');
-        if ((i & 0x0f) == 0x0f)
-        {
-            Serial.println();
-        }
-    }
-    printf("Scan Completed, %d I2C Devices found.\n", cnt);
+void scan() {
+  /* I2C slave Address Scanner
+  for 5V bus
+      * Connect a 4.7k resistor between SDA and Vcc
+      * Connect a 4.7k resistor between SCL and Vcc
+  for 3.3V bus
+      * Connect a 2.4k resistor between SDA and Vcc
+      * Connect a 2.4k resistor between SCL and Vcc
+  */
+  printf("Scanning I2C Addresses:\n");
+  uint8_t cnt = 0;
+  for (uint8_t i = 0; i < 0x80; i++) {
+    Wire.beginTransmission(i);
+    uint8_t ec = Wire.endTransmission(true);
+    if (ec == 0) {
+      printf("%02x ", i);
+      cnt++;
+    } else {
+      printf("-- ");
+    }
+    if ((i & 0x0f) == 0x0f) {
+      printf("\n");
+    }
+  }
+  printf("Scan Completed, %d I2C Devices found.\n", cnt);
```

Comment:
Nice feature, but i would put this into a separate file, e.g. `/lib/I2CBus/I2CBus.cpp`

Reply:
code moved into i2cbus module, call moved to main